crates/wit-bindgen/src/lib.rs

```diff
@@ -27,20 +27,22 @@ mod source;
 mod types;
 use source::Source;
 
-struct InterfaceName {
-    /// True when this interface name has been remapped through the use of `with` in the `bindgen!`
-    /// macro invocation.
-    remapped: bool,
-
-    /// The string name for this interface.
-    path: String,
+#[derive(Clone)]
+enum InterfaceName {
+    /// This interface was remapped using `with` to some other Rust code.
+    Remapped {
+        name_at_root: String,
+        local_path: Vec<String>,
```

Comment:
It looks like this vector only ever has one or three strings in it, would it make sense to make a more precise type that names those cases?

Reply:
I think that's probably a bit overkill on types for now since nothing uses this as anything other than `.join("::")` currently, but if that changes in the future I think it'd be reasonable to add yeah